app/javascript/packages/document-capture/components/document-capture.jsx

```diff
@@ -107,6 +109,14 @@ function DocumentCapture({ isAsyncForm = false, onStepChange }) {
                 })(ReviewIssuesStep)
               : ReviewIssuesStep,
         },
+        {
+          name: 'location',
+          form: InPersonLocationStep,
+        },
```

Comment:
The way this is implemented may present some challenges since I assume we only want the user to progress to the location step if they opt to do IPP. As implemented, the "Submit" behavior of the review step will always proceed to the location step, since it's a linear flow.

Trying to think about how to work around this, ideally it could be nice if we optionally add steps to the flow depending on the user's action (e.g. add the steps when the user opts for IPP), but I dunno if that would be very straight forward.

In other places we've embedded content within the step itself, such as how the [review step](https://github.com/18F/identity-idp/blob/main/app/javascript/packages/document-capture/components/review-issues-step.tsx) includes both the warning page and the form. This works okay, but kinda bloats the step, and requires some careful consideration of page transition logic ([calling `onPageTransition`](https://github.com/18F/identity-idp/blob/2979497b0f60f0acaa042bbf6655742fd9cc5b5c/app/javascript/packages/document-capture/components/review-issues-step.tsx#L83) to ensure accessibility behaviors are retained)

Reply:
So maybe we could have the link to switch to the IPP flow set some value lie `inPerson: true` and make the steps array dynamic based on that value? It sounds to me like finding a way to have dynamic steps would be easier to understand than one additionally-bloated step (and also we'd get a new URL slug for the new step, right?)

Reply:
Yeah, @tomas-nava and I chatted at the end of the day yesterday and came up with a few similar additional ideas to explore:

- Add a condition function in the definition of a step (e.g. `isApplicable(values): boolean`) that allows the step to be skipped
- Allow a step to short-circuit itself and skip to the next step if nothing needs to be done using something similar to what already exists with [`toPreviousStep`](https://github.com/18F/identity-idp/blob/2979497b0f60f0acaa042bbf6655742fd9cc5b5c/app/javascript/packages/form-steps/form-steps.tsx#L440) (e.g. `toNextStep`).
- Track some state at the top-level document capture component which switches between one flow or another for IPP, like [how we do with initial form vs. error review](https://github.com/18F/identity-idp/blob/2979497b0f60f0acaa042bbf6655742fd9cc5b5c/app/javascript/packages/document-capture/components/document-capture.jsx#L96-L122)
  - I think this is what you were suggesting in your first idea?
  - One challenge here is making sure that history and the initial form values aren't lost when we switch into the alternate flow

Reply:
This is addressed in ba91aeac8fa4a4fb298db8a756bc6e365feb0099